### PR TITLE
Add focus state to `ProductItem` expand trigger

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

- Show focus state for the expand button inside `ProductItem`

## Justify why they are needed

- Better accessibility for keyboard users

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #3743** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3743?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #3744** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3744?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
      * **PR #3745** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3745?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #3746** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3746?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3744?utm_source=stack-comment).

**Review by alebedev — APPROVED** [review]

File: apps/store/src/components/ProductItem/ProductDetailsHeader.tsx [hunk 1]
```diff
@@ -40,6 +40,13 @@ const Trigger = styled(Text)({
   alignItems: 'center',
   justifyContent: 'space-between',
   gap: theme.space.xs,
+
+  padding: theme.space.xxxs,
+  margin: `-${theme.space.xxxs}`,
```
Comment (gustaveen) [inline apps/store/src/components/ProductItem/ProductDetailsHeader.tsx:45]: Is this to avoid content shift?

**Review by gustaveen — APPROVED** [review]

File: apps/store/src/components/ProductItem/ProductDetailsHeader.tsx (hunk 1 above)
Comment (robinandeer) [reply apps/store/src/components/ProductItem/ProductDetailsHeader.tsx:45]: i want to show the outline with some padding but i don't want the padding to impact the rest of the layout

**Comment by robinandeer** [comment]
### Merge activity

* **Dec 15, 4:05 PM**: @@robinandeer started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3744).

**Pull request merged**